scripts/linters/js_ts_linter.py

```diff
@@ -429,6 +436,82 @@ def _check_angular_services_index(self) -> concurrent_task_utils.TaskResult:
+        ts_files_to_check = self.ts_filepaths
+        for file_path in ts_files_to_check:
+            comment_before_unknown_type = False
+            multiple_line_comment = False
+            # Not showing lint errors for files present in
+            # FILES_CONTAINING_UNKNOWN_TYPE.
+            if file_path in FILES_CONTAINING_UNKNOWN_TYPE:
+                continue
+
+            file_content = self.file_cache.read(file_path)
+            for line_num, line in enumerate(file_content.split('\n')):
+                unknown_type_indexes: List[int] = []
+                # Check if the line has a comment. Skip the line if it has
+                # a comment.
+                if not (line.strip().startswith('//') or multiple_line_comment):
+                    # Allow the use of 'as unknown' type conversion in spec
+                    # files.
+                    if 'spec' in file_path:
+                        unknown_types = [': unknown', '<unknown']
+                        for unknown_type in unknown_types:
+                            unknown_type_object = re.finditer(
+                                pattern=unknown_type, string=line)
+                            unknown_type_indexes.extend([
+                                index.start() for index in unknown_type_object])
+                    else:
+                        unknown_types = [': unknown', 'as unknown', '<unknown']
+                        for unknown_type in unknown_types:
+                            unknown_type_object = re.finditer(
+                                pattern=unknown_type, string=line)
+                            unknown_type_indexes.extend([
+                                index.start() for index in unknown_type_object])
+
+                # Checking previous line contain comment, if yes then skip throw
+                # errors.
+                if not comment_before_unknown_type:
+                    # Throw error if unknown type is present.
+                    if unknown_type_indexes:
+                        failed = True
+                        for unknown_type_index in unknown_type_indexes:
+                            error_message = (
+                                '%s:%s:%s: unknown type used. Add proper '
+                                'comment explaining why unknown type is '
+                                'used before the line if the type is needed '
+                                'otherwise remove the unknown type and use '
+                                'the appropriate type.' % (
+                                file_path, line_num + 1, unknown_type_index))
+                            error_messages.append(error_message)
+
+                # Check whether previous line contains a proper comment.
+                # If it does, then skip throwing an error.
+                comment_before_unknown_type = (
+                    line.strip().startswith('//') or
+                    line.strip().endswith('*/')
```

Comment:
What about this:

```js
/*
Some comment explaining why we use unknown
*/    // Some other comment
code that uses unknown
```

Reply:
We are not allowing this type of comments in codebase. ESlint throw error first.